python/ray/autoscaler/_private/providers.py

```diff
@@ -11,6 +12,17 @@
 # For caching provider instantiations across API calls of one python session
 _provider_instances = {}
 
+# Minimal config for compatibility with legacy-style external configs.
+MINIMAL_EXTERNAL_CONFIG = {
+    "available_node_types": {
```

Comment:
Instead of calling it head.empty, can we call it `ray-legacy-head-node-type`? It will be confusing for the user to see "empty node types"

Reply:
@DmitriGekhtman , this seems like we missed this comment from Eric.

Reply:
It was originally `head.empty`.
Renaming it `ray-legacy-head-node-type` caused problems that were detected by test_autoscaler.py
so I renamed it `ray.head.default`, which matches the other default configs.

These node types are meant to be overridden anyway
(if they're not, there will be a validation error).
So users won't see these names.